R/req-url.R

```diff
@@ -66,15 +62,20 @@ req_url_path <- function(req, ...) {
 #' @rdname req_url
 req_url_path_append <- function(req, ...) {
   check_request(req)
-  path <- paste(..., sep = "/")
+  path <- dots_to_path(...)
 
   url <- url_parse(req$url)
+  url$path <- paste0(sub("/$", "", url$path), path)
 
+  req_url(req, url_build(url))
+}
+
+dots_to_path <- function(...) {
+  path <- paste(c(...), sep = "/", collapse = "/")
   # Ensure we don't add duplicate /s
-  if (!grepl("^/", path)) {
+  if (path != "" && !grepl("^/", path)) {
```

Comment:
This is necessary b/c `paste(collapse = "/")` returns `""` instead of `character()`.